### DISCUSSION TITLE
Decryption Workflow as a User?

### DISCUSSION THREAD

**Discussion by 0xArch3r** [opentdf/platform#1021] [SDK Q&A] [answered]
I am trying to use the SDK to authenticate to openTDF as a user. When trying to decrypt a TDF Object with an atrtibute applied, I notice the logs show a PERMIT message for jwtentity-1 which is my user who has the attribute, but a DENY for jwtentity-0 which is the opentdf-sdk client I am using. Is this expected? Should both those entities be in the request? How do I add the attributes to the sdk so that it can do what I need it to do if this is the expected behavior?

**Comment by jrschumacher** [comment]
Are you using the token exchange option within the SDK? https://github.com/opentdf/platform/blob/93d8f70750d181e0818911e5b317c9d85044623b/sdk/options.go#L120-L129

**Reply by 0xArch3r** [reply]
I did it with cert exchange and with token exchange. 

We found through testing that the attribute does need to be set through a subject mapping where the client id is the id of the client and that would work. So it appears both the PE and NPE I need to have the attribute

**Comment by ttschampel** [comment, marked as answer]
This is expected behavior until https://github.com/opentdf/platform/issues/871 is decided and implemented to describe scoping and mechanisms to best disambiguate Entity Types

**Reply by jrschumacher** [reply]
@0xArch3r we've decided to use #1181 instead
